### PR TITLE
BE - Create utility to get logged user (DEV only) #37

### PR DESCRIPTION
**Pull request by lenyuk1999oo**

### CODE REVIEW

File: src/main/java/com/chnu/util/LoggedUserUtil.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+import java.util.Optional;
+
+public class LoggedUserUtil {
+
+    public static Optional<UserPrincipal> getLoggedUser() {
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (principal instanceof UserPrincipal) {
+            return Optional.of((UserPrincipal) principal);
+        }
```
Comment (ViktorPelepiak) [inline src/main/java/com/chnu/util/LoggedUserUtil.java]: Are you assured that we must use Optional<> in this place?

Comment (MikeMalyar) [reply src/main/java/com/chnu/util/LoggedUserUtil.java]: Your English is very well :)

Comment (MikeMalyar) [reply src/main/java/com/chnu/util/LoggedUserUtil.java]: But question is good

File: src/main/java/com/chnu/util/LoggedUserUtil.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.chnu.util;
+
+import com.chnu.exception.UserNotFoundException;
+import com.sun.security.auth.UserPrincipal;
+import org.springframework.security.core.context.SecurityContextHolder;
+import java.util.Optional;
+
+public class LoggedUserUtil {
+
+    public static Optional<UserPrincipal> getLoggedUser() {
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/util/LoggedUserUtil.java]: We don't need to return instance of UserPrincipal class. We need to return instance of User class.

File: src/main/java/com/chnu/util/LoggedUserUtil.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (principal instanceof UserPrincipal) {
+            return Optional.of((UserPrincipal) principal);
+        }
+
+        throw new UserNotFoundException("No such user");
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/util/LoggedUserUtil.java]: Is it needful to throw exception there?
For example, what will occur if we call this method for anonymous user?

**Review by MikeMalyar — CHANGES_REQUESTED** [review]

**Review by ViktorPelepiak — APPROVED** [review]

**Review by MikeMalyar — APPROVED** [review]

**Pull request merged**